bus-mapping/src/evm/opcodes/jumpi.rs

```diff
@@ -0,0 +1,122 @@
+use super::Opcode;
+use crate::circuit_input_builder::CircuitInputStateRef;
+use crate::eth_types::GethExecStep;
+use crate::{
+    operation::{StackOp, RW},
+    Error,
+};
+
+/// Placeholder structure used to implement [`Opcode`] trait over it
+/// corresponding to the [`OpcodeId::JUMPI`](crate::evm::OpcodeId::JUMPI)
+/// `OpcodeId`.
+#[derive(Debug, Copy, Clone)]
+pub(crate) struct Jumpi;
+
+impl Opcode for Jumpi {
+    fn gen_associated_ops(
+        state: &mut CircuitInputStateRef,
+        steps: &[GethExecStep],
+    ) -> Result<(), Error> {
+        let step = &steps[0];
+        // `JUMPI` needs two read operation
+        state.push_op(StackOp::new(
+            RW::READ,
+            step.stack.nth_last_filled(0),
+            step.stack.nth_last(0)?,
+        ));
+        state.push_op(StackOp::new(
+            RW::READ,
+            step.stack.nth_last_filled(1),
+            step.stack.nth_last(1)?,
+        ));
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod jumpi_tests {
+    use super::*;
+    use crate::{
+        bytecode,
+        circuit_input_builder::{
+            CircuitInputBuilder, ExecStep, Transaction, TransactionContext,
+        },
+        eth_types::Word,
+        evm::StackAddress,
+        mock,
+    };
+    use pretty_assertions::assert_eq;
+
+    #[test]
+    fn jumpi_opcode_impl() -> Result<(), Error> {
+        let condition = 1;
+        let destination = 35;
```

Comment:
Tho it doesn't affect the zero chunk write in line 63-65, but maybe with a larger value or at least `68` makes more sense?

Reply:
make sense! 9c97d140efc5b67bf0fd4905bde276d8f6d201c4